# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/kwindowsystem/hotfix-bigendian.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- kwindowsystem-6.22.0/src/platforms/xcb/kxutils.cpp.vanilla	2026-01-22 18:34:24.127915898 +0100
+++ kwindowsystem-6.22.0/src/platforms/xcb/kxutils.cpp	2026-01-22 18:39:00.121620691 +0100
@@ -61,11 +61,13 @@
         format = QImage::Format_ARGB32_Premultiplied;
         break;
     default:
+	qDebug() << "NYI fromNative image";
         return T(); // we don't know
     }
     QImage image(xcb_get_image_data(xImage.get()), geo->width, geo->height, xcb_get_image_data_length(xImage.get()) / geo->height, format, free, xImage.get());
     xImage.release();
     if (image.isNull()) {
+	qDebug() << "NYI fromNative image";
         return T();
     }
     if (image.format() == QImage::Format_MonoLSB) {
@@ -87,12 +89,12 @@
 {
 #if Q_BYTE_ORDER == Q_BIG_ENDIAN
     qDebug() << "Byte order not supported";
-    return QPixmap();
+    //return QPixmap();
 #endif
     const xcb_setup_t *setup = xcb_get_setup(c);
     if (setup->image_byte_order != XCB_IMAGE_ORDER_LSB_FIRST) {
         qDebug() << "Byte order not supported";
-        return QPixmap();
+        //return QPixmap();
     }
 
     QPixmap pix = fromNative<QPixmap>(pixmap, c);
